### ISSUE TITLE
Dice loss implementation

### ISSUE THREAD

**Issue by mrgloom** [jocicmarko/ultrasound-nerve-segmentation#85]
I wonder why in dice loss implementation union is just `sum` and not `sum - intersection`, i.e. in case of `sum` seems mask itersection area is calculated twice?

https://github.com/jocicmarko/ultrasound-nerve-segmentation/blob/master/train.py#L27
https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/losses/jaccard.py#L33

**Comment by jizhang02** [comment]
> I wonder why in dice loss implementation union is just `sum` and not `sum - intersection`, i.e. in case of `sum` seems mask itersection area is calculated twice?
> 
> https://github.com/jocicmarko/ultrasound-nerve-segmentation/blob/master/train.py#L27
> https://github.com/keras-team/keras-contrib/blob/master/keras_contrib/losses/jaccard.py#L33

there is another function called Jacard index, you can have a look.

**Comment by jizhang02** [comment]
hello,
I run U-Net using dice loss, but the predicted images are all white. Do you know what's wrong?
def dice_coef(y_true, y_pred):
smooth = 1
y_true_f = K.flatten(y_true)
y_pred_f = K.flatten(y_pred)
intersection = K.sum(y_true_f * y_pred_f)
return (2. * intersection +smooth) / (K.sum(y_true_f) + K.sum(y_pred_f) +smooth)

def dice_coef_loss(y_true, y_pred):
print("dice loss")
return 1-dice_coef(y_true, y_pred)
....
model.compile(optimizer = Adam(lr = 1e-5), loss = dice_coef_loss, metrics = ['accuracy'])
